lapack/gonum/dtrevc3.go

```diff
@@ -647,7 +647,7 @@ leftev:
 					// when forming the right-hand side.
 					beta := math.Max(norms[j], norms[j+1])
 					if beta > vcrit {
-						bi.Dscal(n-ki+1, 1/vmax, b[ki*ldb+iv:], 1)
+						bi.Dscal(n-ki, 1/vmax, b[ki*ldb+iv:], ldb)
```

Comment:
This shows the importance of using also bad matrices for testing. This was started to be covered only after I added zero/tiny eigenvalues to `randomSchurCanonical`.